prov/sm2/src/sm2_av.c

```diff
@@ -45,155 +46,104 @@ static int sm2_av_close(struct fid *fid)
 	if (ret)
 		return ret;
 
-	sm2_map_free(sm2_av->sm2_map);
+	sm2_mmap_unmap_and_close(&sm2_av->sm2_mmap);
+	free(sm2_av->sm2_aux);
 	free(av);
 	return 0;
 }
 
 /*
  * Input address: smr name (string)
- * output address: index (fi_addr_t), the output from util_av
+ * output address: index (fi_addr_t) of the address
+ * @return the number of successful insertions
  */
 static int sm2_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 			 fi_addr_t *fi_addr, uint64_t flags, void *context)
 {
 	struct util_av *util_av;
-	struct util_ep *util_ep;
 	struct sm2_av *sm2_av;
-	struct sm2_ep *sm2_ep;
-	struct dlist_entry *av_entry;
-	fi_addr_t util_addr;
-	int64_t shm_id = -1;
+	int util_addr;
 	int i, ret;
 	int succ_count = 0;
 
 	util_av = container_of(av_fid, struct util_av, av_fid);
 	sm2_av = container_of(util_av, struct sm2_av, util_av);
 
-	for (i = 0; i < count; i++, addr = (char *)addr + strlen(addr) + 1) {
-		FI_INFO(&sm2_prov, FI_LOG_AV, "%s\n", (const char *)addr);
-
-		util_addr = FI_ADDR_NOTAVAIL;
-		if (sm2_av->used < SM2_MAX_PEERS) {
-			ret = sm2_map_add(&sm2_prov, sm2_av->sm2_map, addr, &shm_id);
-			if (!ret) {
-				ofi_mutex_lock(&util_av->lock);
-				ret = ofi_av_insert_addr(util_av, &shm_id, &util_addr);
-				ofi_mutex_unlock(&util_av->lock);
-			}
+	/* acquire the lock */
+	sm2_coordinator_lock(&sm2_av->sm2_mmap);
+
+	for (i = 0; i < count; i++, addr = (char *)addr + SM2_NAME_MAX) {
+		FI_WARN(&sm2_prov, FI_LOG_AV,
+			"fi_av_insert(): About to call sm2_coordinator_allocate_entry() "
+			"on addr %s, my_pid: %d, count: %zu,  \n",
+			(char *)addr, getpid(), count);
+		ret = sm2_coordinator_allocate_entry(addr, &sm2_av->sm2_mmap, &util_addr,
+						     false);
+		FI_WARN(&sm2_prov, FI_LOG_AV,
+			"fi_av_insert(): finished sm2_coordinator_allocate_entry() "
+			"resulting AV Found = %d \n",
+			util_addr);
+		if (ret && util_av->eq) {
+			ofi_av_write_event(util_av, i, -ret, context);
 		}
```

Comment:
Still some lingering format issues to figure out with clang (no {} around single line ifs)

Reply:
https://clang.llvm.org/docs/ClangFormatStyleOptions.html

clang-format 15 (which we use) has a parameter named InsertBraces, which will add braces for all 1 line if's, while's, for's etc... Unfortunately, it doesn't remove braces because most people like the braces (explicit is safer when you add more lines to the if, and easier to read).

I think clang-tidy might be able to do this but I am not sure.

Anyways... if we really care about this, it will have to be done manually from what I can tell.